src/gsAssembler/gsPatchRule.hpp

```diff
@@ -131,7 +132,7 @@ void gsPatchRule<T>::mapTo( const gsVector<T>& lower,
     {
         elNodes[d].resize(m_nodes[d].size());
         elWeights[d].resize(m_weights[d].size());
-        for (auto it = m_maps[d].lower_bound(lower[d]); it!=m_maps[d].upper_bound(upper[d]); it++, k++) // lower_bound = geq, upper_bound= greather than
+        for (auto it = m_maps[d].lower_bound(lower[d]); it!= ( upper[d]==m_end[d] ? m_maps[d].upper_bound(upper[d]) : m_maps[d].lower_bound(upper[d]) ); it++, k++) // lower_bound = geq, upper_bound= greather than
```

Comment:
It is a little better to avoid the upper_bound:

( upper[d]==m_end[d] ? upper[d] : m_maps[d].lower_bound(upper[d]) )

Reply:
Instead of 
`( upper[d]==m_end[d] ? upper[d] : m_maps[d].lower_bound(upper[d]) )`
It should have been
`( upper[d]==m_end[d] ? m_maps[d].end() : m_maps[d].lower_bound(upper[d]) )`